Document compat flag: fetcher_no_get_put_delete (#13288)
cloudflare/cloudflare-docs: content/workers/_partials/_platform-compatibility-dates/fetcher-no-get-put-delete.md

---
_build:
  publishResources: false
  render: never
  list: never

> name: "Fetchers no longer have get/put/delete helper methods"

Reviewer: Not worried about since few if any people should really need to dig into this flag, but (AFAIK) we haven't referred to this in public docs as "fetcher" in the past?

```suggestion
name: "Removes built-in get/put/delete helper methods"
```

Author: I debated this myself but it seems at least somewhat intuitive to say that objects which implement `fetch()` are "Fetchers", and I didn't want to make this title overly verbose by saying "Durable Object stubs and Service Bindings".

I worry that your suggestion here might confuse people as they might think of other places in our API that have such method names, like KV namespaces or DO storage.

Reviewer: that's fair — ship it!

sort_date: "2024-03-26"
enable_date: "2024-03-26"
enable_flag: "fetcher_no_get_put_delete"
disable_flag: "fetcher_has_get_put_delete"
---

> Some objects in the Workers API implement a method `fetch()` which behaves similarly to the global `fetch()` method, but the requests are sent to the destination represented by the object, rather than being routed based on the URL. In particular, [Durable Object](/durable-objects/) stubs and [Service Bindings](/workers/configuration/bindings/about-service-bindings/) both have such `fetch()` methods.

Reviewer: ```suggestion
[Durable Object](/durable-objects/) stubs and [Service Bindings](/workers/configuration/bindings/about-service-bindings/) both implement a `fetch()` method which behaves similarly to the global `fetch()` method, but requests are instead sent to the destination represented by the object, rather than being routed based on the URL.
```

> Some objects in the Workers API implement a method `fetch()` which behaves similarly to the global `fetch()` method, but the requests are sent to the destination represented by the object, rather than being routed based on the hostname. In particular, [Durable Object](https://developers.cloudflare.com/durable-objects/) stubs and [Service Bindings](https://developers.cloudflare.com/workers/configuration/bindings/about-service-bindings/) both have such `fetch()` methods.

Author: ```suggestion
Some objects in the Workers API implement a method `fetch()` which behaves similarly to the global `fetch()` method, but the requests are sent to the destination represented by the object, rather than being routed based on the URL. In particular, [Durable Object](/durable-objects/) stubs and [Service Bindings](/workers/configuration/bindings/about-service-bindings/) both have such `fetch()` methods.
```

Historically, API objects that had such a `fetch()` method also had methods `get()`, `put()`, and `delete()`. These methods were thin wrappers around `fetch()` which would perform the corresponding HTTP method and automatically handle writing/reading the request/response bodies as needed.

> These methods were thought to be a neat idea but were never actually documented. In retrospect, they don't seem so great anymore. Moreover, they may soon get in the way: we would like for applications to define their own methods on these types, and applications may be surprised if they cannot define `get`, `put`, and `delete` due to conflicting with these helpers.

Reviewer: ```suggestion
These methods were a very early idea from many years ago, but were never actually documented, and therefore rarely (if ever) used. Enabling the `fetcher_no_get_put_delete`, or setting a compatibility date on or after `2024-03-26` disables these methods for your Worker.

This change paves a future path for you to be able to define your own custom methods using these names. Without this change, you would be unable to define your own `get`, `put`, and `delete` methods, since they would conflict with these built-in helper methods.
```

> Therefore, these methods are being removed. Since they were never documented, it's unlikely anyone relies on them.

Reviewer: ```suggestion
```
